cfme/tests/infrastructure/test_tenant_quota.py

```diff
@@ -77,3 +95,35 @@ def test_tenant_quota_enforce(appliance, provider, setup_provider, set_roottenan
         provision_request.approve_request(method='ui', reason="Approved")
     provision_request.wait_for_request(method='ui')
     assert provision_request.row.reason.text == "Quota Exceeded"
+
+
+# first arg of parametrize is the list of fixtures or parameters,
+# second arg is a list of lists, with each one a test is to be generated
+# sequence is important here
+# indirect is the list where we define which fixtures are to be passed values indirectly.
+@pytest.mark.parametrize(
+    ['set_roottenant_quota', 'custom_prov_data', 'extra_msg'],
+    [
+        [('cpu', 2), {'hardware': {'num_sockets': '8'}}, ''],
+        [('storage', 0.01), {}, ''],
+        [('memory', 2), {'hardware': {'memory': '4096'}}, ''],
+        [('vm', 1), {'catalog': {'num_vms': '4'}}, '###']
+    ],
+    indirect=['set_roottenant_quota'],
+    ids=['max_cpu', 'max_storage', 'max_memory', 'max_vms']
+)
+def test_tenant_quota_enforce_via_service(appliance, provider, setup_provider,
+                                          set_roottenant_quota, extra_msg, custom_prov_data,
+                                          prov_data, template_name, catalog_item):
+    """Test Tenant Quota in UI"""
+    catalog_item.provisioning_data.update(custom_prov_data)
+    catalog_item.provisioning_data['catalog']['vm_name'] = catalog_item.vm_name
+    catalog_item.create()
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog,
+                                       catalog_item.name)
+    service_catalogs.order()
+    # nav to requests page to check quota validation
+    request_description = 'Provisioning Service [{0}] from [{0}]'.format(catalog_item.name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
```

Comment:
**REQUIRED**
Please also remove the request as well using `addfinalizer`

Reply:
added.